### ISSUE TITLE
Handling of Predictor responses

### ISSUE THREAD

**Issue by Pirol62** [dadaloop82/HASS_AppDeamon_SwitchPredictor#1]
I would implement it like this: the event triggers an automation, which let alexa/google ask: You switched this entity by 90% during the last week. Do you want to switch it now? if yes a second question should be “shall I create an automation?” When yes, alexa never have to ask again.

A second issue comes up when thinking about a yearly time period. The behavior changes i.e. over the seasons. In summertime a switch is changed at another time or never compared to winter. So the prediction has to be calculated over a long (at least a year) time period and will provide different results over this time.
So my first suggestion, creating an automation, is not that good in a first state. Otherwise: Let alexa always ask questions can lead to annoying situations where alexa asks too often. Maybe, there should be the option to reduce this support only to a type of switches (light, covers, heating etc)

**Comment by dadaloop82** [comment]
> I would implement it like this: the event triggers an automation, which let alexa/google ask: You switched this entity by 90% during the last week. Do you want to switch it now? if yes a second question should be “shall I create an automation?” When yes, alexa never have to ask again.

It's a good idea, but in my opinion, by doing so, you make everything too static by actually creating a routine that then if the conditions change can no longer be modified.
I was thinking these days to have Home Assistant not create routines "automatically", for this very reason, but to have HASS_AppDeamon_SwitchPredictor handle the issue, also based on the responses that come back from Alexa (via Home Assistant) that can strengthen or weaken the eventual probability.

> A second issue comes up when thinking about a yearly time period. The behavior changes i.e. over the seasons. In summertime a switch is changed at another time or never compared to winter. So the prediction has to be calculated over a long (at least a year) time period and will provide different results over this time.
> So my first suggestion, creating an automation, is not that good in a first state. Otherwise: Let alexa always ask questions can lead to annoying situations where alexa asks too often. Maybe, there should be the option to reduce this support only to a type of switches (light, covers, heating etc)

So, currently I manage, in the probabilistic calculation: the minutes, the hours, the day, the month and the day of the week (Monday, Tuesday, etc...) and it's a great idea to manage also the seasons, but also other variables (ex: the year etc) - probably I will manage everything as "virtual" sensors - but I still have to think about it

Yes, letting Alexa ask too many questions is really disturbing, but in the beginning unfortunately it will be unavoidable for a short period of time - the theory is that the system learns based on probabilistic calculations and based on the answers given via Alexa/Home Assistant and determines a weight.

Currently as "basic" sensors only switch type (on/off) are supported because they are obviously easier to control, but as "correlated" sensors (basedON) all types of sensors are supported, booelane, string, binary, etc. ...

**Comment by Pirol62** [comment]
That sounds good. 
One additonal thing: don't forget to create an option to swich off the prediction for certain devices. Once an automation is active, the prediction may not relevant for the user.

**Comment by dadaloop82** [comment]
Hi!
The prediction is based on a basic switch and connected sensors (string, integer, booelean, etc...) that you set in the configuration.

example:

```
predictsevent:
      turontelevision:
        basewitch: switch.samsung_8_series_50
        basedon:
          [
            person.daniel_2,
            person.giuliana,
            sensor.temperatura_e_umidita_temperature_measurement,
            sensor.temperatura_e_umidita_relative_humidity_measurement,
          ]
```

It means that for the <turontelevision> event, the basic switch is <switch.samsung_8_series_50> correlated to <person.daniel_2>, <person.giuliana>, <sensor.temperature_and_humidity_temperature_measurement> and <sensor.temperature_and_humidity_relative_humidity_measurement>.

**Also, I have some great news!**

I've updated the data model by taking into account, when learning from history, the day of the week and the season! 
Thanks for your precious suggestion, I've also updated the readme (the project description)

I also added the possibility to offer me a beer or a coffee, if you like to support me also economically and give me more and more desire to continue the project :) :) :)

Thanks for all !!
